src/rpc/rpc-handler.ts

```diff
@@ -103,7 +103,9 @@ export class RPCHandler {
         callback(EVENT.CLIENT_OFFLINE)
         return
       }
-      return Promise.reject(EVENT.CLIENT_OFFLINE)
+      return new Promise((resolve, reject) => {
+        nextTick(reject.bind(this, EVENT.CLIENT_OFFLINE))
+      })
```

Comment:
also i think promises don't resolve synchronously, so this is unnecessary
https://jsblog.insiderattack.net/promises-next-ticks-and-immediates-nodejs-event-loop-part-3-9226cbe7a6aa